Makefile

```
@@ -17,24 +17,136 @@
 # limitations under the License.
 
 
+ifeq "$(LC_CTYPE)" ""
+	LC_CTYPE="en_US.UTF-8"
+endif
+
+ifndef SHELL
+	SHELL:=command -pv bash
+endif
+
+ifeq "$(ERROR_LOG_PATH)" ""
+	ERROR_LOG_PATH="/dev/null"
+endif
+
+ifeq "$(COMMAND)" ""
+	COMMAND_CMD=$(shell command -v xcrun || command -v command || command which which || command -v which)
+	COMMAND_TOOL=$(notdir $(COMMAND_CMD))
+	ifeq "$(COMMAND_TOOL)" "xcrun"
+		COMMAND_ARGS= --find
+	endif
+	ifeq "$(COMMAND_TOOL)" "command"
+		COMMAND_ARGS= -v
+	endif
+	ifeq "$(COMMAND_CMD)" ""
+		COMMAND_CMD="command"
+	endif
+	ifdef COMMAND_ARGS
+		COMMAND := $(COMMAND_CMD) $(COMMAND_ARGS)
+	else
+		COMMAND := $(COMMAND_CMD)
+	endif
+endif
+
+# Check if POSIX touch command can be resolved by COMMAND in the runtime env
+ifeq "$(notdir $(shell $(COMMAND) touch))" ""
+	# This is a non-POSIX environment, so try Windows fallback logic
```

:thinking: the comment here is actually a little misleading; Let's reword this slightly for accuracy.

```suggestion
	# This is a non-POSIX environment, so try GHA-Windows-Latest Specific fallback logic
```